### PR TITLE
TestTask exec unit tests, fixes #270

### LINKED ISSUE #270: Unit test check fails if project has only one unit test
Relatively minor issue, but caused me a bit of grief getting started with the plugin :)

In the TestTask line 134 the regex checks for the following:
```
int testCount = Utils.matchNumberRegex(outputStr, /OK \((\d+) tests\)/)
```
However if you have only one unit test the success message would be:

OK (1 test)

Which in turn fails the build.

### PR DESCRIPTION
**Pull request by brunobowden**
- Refactor testSetup() code for TestTask

### CODE REVIEW

**Comment by brunobowden** [comment]
@alexmetcalfe - here's your fix. It's not even a 1 line fix as it's only a 1 character fix! Along with 50 lines of testing that needed to be added anyway. Not sure when 0.4 comes but look for that.

@advayDev1, @confile - feel free to look as well

File: src/test/groovy/com/github/j2objccontrib/j2objcgradle/tasks/TestTaskTest.groovy [hunk 1]
```diff
@@ -81,4 +88,71 @@ class TestTaskTest {
 
         assert expectedTestNames == testNames
     }
+
+    private void taskSetup() {
```
Comment (advayDev1) [inline src/test/groovy/com/github/j2objccontrib/j2objcgradle/tasks/TestTaskTest.groovy]: verb first - setupTask

File: src/test/groovy/com/github/j2objccontrib/j2objcgradle/tasks/TestTaskTest.groovy [hunk 2]
```diff
@@ -81,4 +88,71 @@ class TestTaskTest {
...
+        // For Utils.throwIfNoJavaPlugin()
+        proj.pluginManager.apply(JavaPlugin)
+
+        // For Utils.J2objcHome()
+        j2objcHome = File.createTempDir('J2OBJC_HOME', '').path
```
Comment (advayDev1) [inline src/test/groovy/com/github/j2objccontrib/j2objcgradle/tasks/TestTaskTest.groovy]: i've seen this in some of your other testing code too.  maybe refactor to a TestingUtils class?

**Comment by advayDev1** [comment]
LGTM with minor comments

File: src/test/groovy/com/github/j2objccontrib/j2objcgradle/tasks/TestTaskTest.groovy (hunk 1 above)
Comment (brunobowden) [inline src/test/groovy/com/github/j2objccontrib/j2objcgradle/tasks/TestTaskTest.groovy]: fixed

**Comment by brunobowden** [comment]
I've refactored to a TestingUtils class the shared code.

**Pull request merged**